2-servers/src/main/scala/com/snowplowanalytics/piinguin/server/clients/DynamoDBClient.scala

```diff
@@ -0,0 +1,113 @@
+
+// Generated
+import com.snowplowanalytics.piinguin.server.generated.protocols.piinguin._
+
+/**
+ * Piinguin Client for dynamo
+ * @param client the amazon dynamo client to be used
+ * @param tableName the name of the table to be used (it must exist)
+ */
+class DynamoDBClient(val client: AmazonDynamoDBAsync, val tableName: String)(implicit ec: ExecutionContext) {
+  private val table: Table[PiiRecord] = Table[PiiRecord](tableName)
+
+  /**
+   * Write a single record asynchronously
+   * @param record PiiRecord object to be written
+   * @return a future for the execution of the request with either an error or the odl record if replaced
+   */
+  def putRecord(record: PiiRecord): Future[Option[Either[DynamoReadError, PiiRecord]]] = exec(table.put(record))
+
+  /**
+   * Get a single record asynchronously
+   * @param modifiedValue the value for the table key
+   * @return a future for the execution of the request with either an error or the record if found
+   */
+  def getRecord(modifiedValue: String): Future[Option[Either[DynamoReadError, PiiRecord]]] =
+    exec(table.get('modifiedValue -> modifiedValue))
+
+  /**
+   * Delete a single record asynchronously
+   * @param modifiedValue the value for the table key
+   * @return a future for the execution of the request with the result of the deletion
+   */
+  def deleteRecord(modifiedValue: String): Future[DeleteItemResult] =
+    exec(table.delete('modifiedValue -> modifiedValue))
+
+  /**
+   * Write mutltiple records asynchronously
+   * @param records List of PiiRecord objects to be written
+   * @return a future with the result of the batch write
+   */
+  def putRecords(records: List[PiiRecord]): Future[List[BatchWriteItemResult]] = exec(table.putAll(records.toSet))
+
+  /**
+   * Get mutltiple records asynchronously
+   * @param records the list of requests containing the modifiedValues to be read
+   * @return a future for the execution of the request with a set of either an error or the record if found
+   */
+  def getRecords(records: List[ReadPiiRecordRequest]): Future[Set[Either[DynamoReadError, PiiRecord]]] =
+    exec(table.getAll('modifiedValue -> records.map(_.modifiedValue).toSet))
+
+  /**
+   * Delete mutltiple records asynchronously
+   * @param records the list of requests containing the modifiedValues to be deleted
+   * @return a future for the execution of the request with a list of the results
+   */
+  def deleteRecords(records: List[DeletePiiRecordRequest]): Future[List[BatchWriteItemResult]] =
+    exec(table.deleteAll('modifiedValue -> records.map(_.modifiedValue).toSet))
+
+  private def exec[A](ops: ScanamoOps[A]): Future[A] = ScanamoAsync.exec(client)(ops)
+
+}
+
+/**
+ * Normal client for dynamodb
+ */
+object DynamoDBClient {
+  lazy val client = AmazonDynamoDBAsyncClientBuilder.defaultClient()
+}
+
+/**
+ * Test client for dynamodb using default test end point
+ */
+object DynamoDBTestClient {
+  def client =
```

Comment:
why is this a def?

Reply:
So that you get a new client every time you call it..

Reply:
yes so the question should be why do you need a new client every time?

Reply:
I changed this.. True makes no sense